### ISSUE TITLE
EOS, SOS characters in dataloader

### ISSUE THREAD

**Issue by paanguin** [gentaiscool/end2end-asr-pytorch#33]
I have a question about decoder inputs. I think the following pre-processing adds SOS and EOS token to label y. 
https://github.com/gentaiscool/end2end-asr-pytorch/blob/a22efdda28f1689206eefb3fedbb56fbcc98a753/models/asr/transformer.py#L261
https://github.com/gentaiscool/end2end-asr-pytorch/blob/a22efdda28f1689206eefb3fedbb56fbcc98a753/models/asr/transformer.py#L262

It seems `SpectrogramDataset`  also contain a process for adding SOS and EOS to label y.
https://github.com/gentaiscool/end2end-asr-pytorch/blob/a22efdda28f1689206eefb3fedbb56fbcc98a753/utils/data_loader.py#L137
But I think `SpectrogramDataset` should not do this. I think the decoder currently process the label like this:

`seq_in`: SOS, SOS, y[0], y[1], ... y[T-1], EOS
`seq_out`: SOS, y[0], y[1], ... y[T-1], EOS, EOS

I'll be very grateful if you confirm whether this is correct or not.

**Comment by gentaiscool** [comment]
Thank you for reporting this issue @paanguin. Actually, we are working on revamping the project with a newer version soon.

Yes, I agree we need to fix this part. There should be a single SOS and EOS tokens. We should not include SOS and EOS tokens in SpectrogramDataset. We will fix this issue in the next update.
